### DISCUSSION TITLE
Can I get this via the API?

### DISCUSSION THREAD

**Discussion by JPhilipp** [thygate/stable-diffusion-webui-depthmap-script#24] [Q&A]
Hi, great script -- how would one get the depth map for a given input image when using the Automatic1111 local API (as I do in Unity)? Thanks!

**Comment by thygate** [comment]
I've not had the chance to check the API myself, but it works in krita, so it might be supported .. (no options though for my script, in krita .. it also lists script support as experimental)

![image](https://user-images.githubusercontent.com/96586282/202015698-7b5d940d-bd20-4148-8e61-746a8a2bdc42.png)

If you only want to generate a depthmap from an existing image, you could use img2img and set denoising strength to zero, this will effectively skip stable diffusion and generate the depthmap from the input image.

I will look into it ..

**Reply by thygate** [reply]
After checking the krita plugin frontend, it adds the elements `script` and `script_args` to the request json.
But if I try those in http://localhost:7860/docs#/default/text2imgapi_sdapi_v1_txt2img_post it doesn't seem to be supported.. Krita also uses its own backend, not the v1 API, and also includes some hacks to get the script options from the gradio elements ..

Looking further through the v1 API, I cannot find support for scripts. So it is not supported at this time.
